src/components/Form/FileUpload/FileUpload.test.tsx

```diff
@@ -347,6 +387,59 @@ describe("upload action", () => {
     expect(onDrop).toHaveBeenCalledTimes(0);
   });
 
+  it("show error state and error message on drop of invalid extension file", async () => {
+    const onDrop = jest.fn();
+    const { container } = createComponent(
+      defaultParams => ({
+        ...defaultParams,
+        onDrop,
+        fileList: []
+      }),
+      "file-upload-6"
+    );
+
+    const file = new File([""], "test.png", {
+      type: "image/png"
+    });
+
+    const eventData = {
+      dataTransfer: {
+        files: [file]
+      }
+    };
+
+    const dropZone = container.querySelector(".upload-button-wrapper") as Element;
+    const dropEvent = createEvent.drop(dropZone, eventData);
+
+    await waitFor(() => {
+      fireEvent(dropZone, dropEvent);
+    });
+
+    const dragDropWrapper = container.querySelector(".file-dropzone") as Element;
+    const errorMessage = container.querySelector(".file-selector-sub-text") as Element;
+
+    expect(dragDropWrapper.classList.contains("error")).toBeTruthy();
+    expect(errorMessage.classList.contains("error")).toBeTruthy();
+    expect(errorMessage.innerHTML).toStrictEqual(
+      `Invalid file format. Supported formats are: ${defaultParams.accept}.`
+    );
+  });
+
+  it("sub text or custom text is visible", async () => {
+    const { container } = createComponent(
+      defaultParams => ({
+        ...defaultParams,
+        fileList: [],
+        subText: "This is sub heading."
+      }),
+      "file-upload-6"
+    );
+    const subTextElement = container.querySelector(".file-selector-sub-text") as Element;
+
+    expect(subTextElement).not.toBeNull();
+    expect(subTextElement.innerHTML).toStrictEqual("This is sub heading.");
```

Comment:
```suggestion
    expect(subTextElement).toHaveTextContent("This is sub heading.");
```
this is a bit better to read imo

Reply:
`toStrictEqual` check string with exact/strictly but `toHaveTextContent` is not a strict equality check

Reply:
infact if there would be any other or more text coming then still it will pass test case because `toHaveTextContent` just checks provided string is available or not. No matter what exactly text is coming. For Ex: It will pass below both string cases:
1.  This is sub heading.
2. This is sub heading. I have added more text here.